cmd/thanos/receive.go

```diff
@@ -77,9 +77,11 @@ func registerReceive(app *extkingpin.App) {
 			MaxExemplars:           conf.tsdbMaxExemplars,
 		}
 
-		// Enable ingestion if endpoint is specified or if both the hashrings configs are empty.
-		// Otherwise, run the receiver exclusively as a distributor.
-		enableIngestion := conf.endpoint != "" || (conf.hashringsFileContent == "" && conf.hashringsFilePath == "")
+		// Has the user provided some kind of hashring configuration?
+		hashringSpecified := conf.hashringsFileContent != "" || conf.hashringsFilePath != ""
+		// Has the user specified the --receive.local-endpoint flag?
+		localEndpointSpecified := conf.endpoint != ""
+		receiveMode := receive.DetermineMode(hashringSpecified, localEndpointSpecified)
```

Comment:
If we do this complex thing of determining, why not just passing `config`? Sounds like related complexity we can hide (reader of code has  to visit this thing anyway)

Reply:
yeah that would simplify things - nice idea